src/ccl_background.c

```diff
@@ -434,153 +452,176 @@ void ccl_cosmology_compute_distances(ccl_cosmology * cosmo, int *status)
     return;
   }
 
-  // Create logarithmically and then linearly-spaced values of the scale factor
-  int na = cosmo->spline_params.A_SPLINE_NA+cosmo->spline_params.A_SPLINE_NLOG-1;
-  double * a = ccl_linlog_spacing(
-    cosmo->spline_params.A_SPLINE_MINLOG, cosmo->spline_params.A_SPLINE_MIN,
-    cosmo->spline_params.A_SPLINE_MAX, cosmo->spline_params.A_SPLINE_NLOG,
-    cosmo->spline_params.A_SPLINE_NA);
-  // Allocate arrays for all three of E(a), chi(a), and a(chi)
-  double *E_a = malloc(sizeof(double)*na);
-  double *chi_a = malloc(sizeof(double)*na);
-  // Allocate E(a) and chi(a) splines
-  gsl_spline * E = gsl_spline_alloc(cosmo->spline_params.A_SPLINE_TYPE, na);
-  gsl_spline * chi = gsl_spline_alloc(cosmo->spline_params.A_SPLINE_TYPE, na);
-  // a(chi) spline allocated below
-
-  //Check for too little memory
-  if (a==NULL || E_a==NULL || chi_a==NULL){
-    *status=CCL_ERROR_MEMORY;
-
-    ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_cosmology_compute_distances(): ran out of memory\n");
-  }
-
-  //Check for messed up scale factor conditions
-  if (!*status){
-    if ((fabs(a[0]-cosmo->spline_params.A_SPLINE_MINLOG)>1e-5) ||
-	(fabs(a[na-1]-cosmo->spline_params.A_SPLINE_MAX)>1e-5) ||
-	(a[na-1]>1.0)) {
-      *status = CCL_ERROR_LINSPACE;
-      ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_cosmology_compute_distances(): Error creating first logarithmic and then linear spacing in a\n");
+  #pragma omp master
```

Comment:
I thought these functions were only gonna ever be called from python (i.e. always outside of openmp loops). If that's the case, why the pragma?

Reply:
Hold overs from previous versions of this. Happy to remove. They don’t do any harm here.